pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java

```diff
@@ -507,8 +510,8 @@ public void redeliverUnacknowledgedMessages() {
         flowConsumerBlockedPermits(this);
         if (pendingAcks != null) {
             int totalRedeliveryMessages = 0;
-            for (Integer batchSize : pendingAcks.values()) {
-                totalRedeliveryMessages += batchSize;
+            for (LongPair batchSize : pendingAcks.values()) {
```

Comment:
maybe use `AtomicInteger` to accumulate since you cannot just add over the same `int` from a lambda